src/java.base/share/classes/java/lang/reflect/AccessFlag.java

```diff
@@ -0,0 +1,420 @@
+     * <code>{@value "0x%04x" Modifier#STRICT}</code>.
+     */
+    STRICT(Modifier.STRICT, true, Set.of(Location.METHOD)),
+
+    /**
+     * The access flag {@code ACC_SYNTHETIC} with a mask value of
+     * <code>{@value "0x%04x" Modifier#SYNTHETIC}</code>.
+     * @see Class#isSynthetic()
+     * @see Executable#isSynthetic()
+     * @see java.lang.module.ModuleDescriptor.Modifier#SYNTHETIC
+     */
+    SYNTHETIC(Modifier.SYNTHETIC, false,
+              Set.of(Location.CLASS, Location.FIELD, Location.METHOD,
+                     Location.INNER_CLASS, Location.METHOD_PARAMETER,
+                     Location.MODULE, Location.MODULE_REQUIRES,
+                     Location.MODULE_EXPORTS, Location.MODULE_OPENS)),
+
+    /**
+     * The access flag {@code ACC_ANNOTATION} with a mask value of
+     * <code>{@value "0x%04x" Modifier#ANNOTATION}</code>.
+     * @see Class#isAnnotation()
+     */
+    ANNOTATION(Modifier.ANNOTATION, false,
+               Set.of(Location.CLASS, Location.INNER_CLASS)),
+
+    /**
+     * The access flag {@code ACC_ENUM} with a mask value of
+     * <code>{@value "0x%04x" Modifier#ENUM}</code>.
+     * @see Class#isEnum()
+     */
+    ENUM(Modifier.ENUM, false,
+         Set.of(Location.CLASS, Location.FIELD, Location.INNER_CLASS)),
+
+    /**
+     * The access flag {@code ACC_MANDATED} with a mask value of
+     * <code>{@value "0x%04x" Modifier#MANDATED}</code>.
+     */
+    MANDATED(Modifier.MANDATED, false,
+             Set.of(Location.METHOD_PARAMETER,
+                    Location.MODULE, Location.MODULE_REQUIRES,
+                    Location.MODULE_EXPORTS, Location.MODULE_OPENS)),
+
+    /**
+     * The access flag {@code ACC_MODULE} with a mask value of {@code
+     * 0x8000}.
+     */
+    MODULE(0x0000_8000, false, Set.of(Location.CLASS))
+    ;
+
+    // May want to override toString for a different enum constant ->
+    // name mapping.
+
+    private int mask;
+    private boolean sourceModifier;
+
+    // Intentionally using Set rather than EnumSet since EnumSet is
+    // mutable.
+    private Set<Location> locations;
+
+    private AccessFlag(int mask, boolean sourceModifier, Set<Location> locations) {
+        this.mask = mask;
+        this.sourceModifier = sourceModifier;
+        this.locations = locations;
+    }
+
+    /**
+     * {@return the corresponding integer mask for the access flag}
+     */
+    public int mask() {
+        return mask;
+    }
+
+    /**
+     * {@return whether or not the flag has a directly corresponding
+     * modifier in the Java programming language}
+     */
+    public boolean sourceModifier() {
+        return sourceModifier;
+    }
+
+    /**
+     * {@return kinds of constructs the flag can be applied to}
+     */
+    public Set<Location> locations() {
+        return locations;
+    }
+
+    /**
+     * {@return a set of access flags for the given mask value
+     * appropriate for the location in question}
```

Comment:
Should we specify that the returned set is unmodifiable/immutable?

Reply:
Sure; that would be consistent with other parts of the PR.